packages/xchain-ethereum/src/client.ts

```diff
@@ -365,15 +361,41 @@ export default class Client implements XChainClient, EthereumClient {
    * Get the transaction details of a given transaction id.
    *
    * @param {string} txId The transaction id.
-   * @param {string} asset The asset address. (optional)
+   * @param {string} assetAddress The asset address. (optional)
    * @returns {Tx} The transaction details of the given transaction id.
    *
    * @throws {"Need to provide valid txId"}
    * Thrown if the given txId is invalid.
    */
-  getTransactionData = async (txId: string, asset?: string): Promise<Tx> => {
+  getTransactionData = async (txId: string, assetAddress?: string): Promise<Tx> => {
     try {
-      const tx = await etherscanAPI.getTransactionData(this.etherscan.baseUrl, txId, asset, this.etherscan.apiKey)
+      let tx
+      const txInfo = await this.etherscan.getTransaction(txId)
+
+      if (txInfo) {
+        if (assetAddress) {
+          tx =
+            (
+              await etherscanAPI.getTokenTransactionHistory({
+                baseUrl: this.etherscan.baseUrl,
+                assetAddress,
+                startblock: txInfo.blockNumber,
+                endblock: txInfo.blockNumber,
+                apiKey: this.etherscan.apiKey,
+              })
+            ).find((info) => info.hash === txId) ?? null
+        } else {
+          tx =
+            (
+              await etherscanAPI.getTokenTransactionHistory({
+                baseUrl: this.etherscan.baseUrl,
+                startblock: txInfo.blockNumber,
+                endblock: txInfo.blockNumber,
+                apiKey: this.etherscan.apiKey,
+              })
+            ).find((info) => info.hash === txId) ?? null
+        }
+      }
```

Comment:
One `etherscanAPI.getTokenTransactionHistory` does its job here, there is no difference for both `await etherscanAPI.getTokenTransactionHistory(...)` calls ^. Missing or invalid `assetAddress` is already handled by [getTokenTransactionHistory](https://github.com/xchainjs/xchainjs-lib/blob/d876665bc4677d4662272fc5e032a754250c080e/packages/xchain-ethereum/src/etherscan-api.ts#L95)